### PR TITLE
Feature/ai 175 automatic and manual renumber

### PR DESCRIPTION
**Pull request by PuertolasEngworks**
Added automatic renumber
Redesign UI

### CODE REVIEW

File: RenumberParts_Shared/Core/RenumberMain.cs [hunk 1]
```diff
@@ -191,6 +191,178 @@ public static void AddToSelection()
...
+        /// <summary>
+        /// Save an element on a temporary list and override its color
+        /// </summary>
+        public static void AddToSelection(ElementId ReferenceElem, List<Element> completeList)
+        {
+
+                foreach (Element elem in completeList)
+                {
+                selectedElements.Add(elem);
+                //Category cat = elem.Category;
```
Comment (pabloderen) [inline RenumberParts_Shared/Core/RenumberMain.cs]: Remove commented code

File: RenumberParts_Shared/Core/RenumberMain.cs [hunk 2]
```diff
@@ -191,6 +191,178 @@ public static void AddToSelection()
...
+            foreach (Element x in selectedElements)
+            {
+                //Override color of element
+                doc.ActiveView.SetElementOverrides(x.Id, overrideGraphicSettings);
+            }
+
+        }
+
+
+        public static bool filterParam(Element elementEx, Element elementExP, BuiltInParameter param01, BuiltInParameter param02,
```
Comment (pabloderen) [inline RenumberParts_Shared/Core/RenumberMain.cs]: Camel case Method

File: RenumberParts_Shared/Views/MainForm.xaml [hunk 3]
```diff
@@ -5,7 +5,7 @@
              xmlns:d="http://schemas.microsoft.com/expression/blend/2008" 
         xmlns:sw="clr-namespace:WpfStyleableWindow.StyleableWindow"      
              xmlns:local="clr-namespace:RenumberParts"
-        mc:Ignorable="d" Height="445" Width="280" MinHeight="442" MinWidth="280"  MaxHeight="430" MaxWidth="280" WindowStartupLocation="CenterScreen"
+        mc:Ignorable="d" Height="385.667" Width="280" MinHeight="442" MinWidth="280"  MaxHeight="430" MaxWidth="280" WindowStartupLocation="CenterScreen"
```
Comment (pabloderen) [inline RenumberParts_Shared/Views/MainForm.xaml]: Round size to neares 100 or 10 px

**Review by pabloderen — CHANGES_REQUESTED** [review]

File: RenumberParts_Shared/Views/MainForm.xaml.cs [hunk 4]
```diff
@@ -151,10 +171,540 @@ private void AddButton_Click(object sender, RoutedEventArgs e)
...
+                }
+            }
+            return tempConnectors;
+        }
+
+        /// <summary>
+        /// Get The connector set of an element
+        /// </summary>
+        /// <param name="elem"></param>
+        private int getElementConected(ref Element elem)
```
Comment (pabloderen) [inline RenumberParts_Shared/Views/MainForm.xaml.cs]: Change name to **getElementConnected**

File: RenumberParts_Shared/Views/MainForm.xaml.cs [hunk 5]
```diff
@@ -151,10 +171,540 @@ private void AddButton_Click(object sender, RoutedEventArgs e)
...
+                elemIds.RemoveAt(elemIds.Count() - 1);
+                System.Windows.MessageBox.Show("The elements has more than one conection", "warning");
+            }
+
+                while (countElem==1)
+            {
+                countElem = getElementConected(ref elem);
+            }
+
+            //ConnectorSet connectors = this.getConnectorSetFromElement(elem);
```
Comment (pabloderen) [inline RenumberParts_Shared/Views/MainForm.xaml.cs]: Remove commented code

File: RenumberParts_Shared/Views/MainForm.xaml.cs [hunk 6]
```diff
@@ -151,10 +171,540 @@ private void AddButton_Click(object sender, RoutedEventArgs e)
...
 
+        /// <summary>
+        /// Add spool number to elements
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void AutoButton_Click(object sender, RoutedEventArgs e)
+        {
+
+            Connector CurrentConnector = null;
```
Comment (pabloderen) [inline RenumberParts_Shared/Views/MainForm.xaml.cs]: Variable not being used

File: RenumberParts_Shared/Views/MainForm.xaml.cs [hunk 7]
```diff
@@ -151,10 +171,540 @@ private void AddButton_Click(object sender, RoutedEventArgs e)
...
+            return tempConnectors;
+        }
+
+        /// <summary>
+        /// Get The connector set of an element
+        /// </summary>
+        /// <param name="elem"></param>
+        private int getElementConected(ref Element elem)
+        {
+            List<string> categorias = new List<string>(new string[] { "OST_CableTrayFitting", "OST_CableTray", "OST_ConduitFitting", "OST_Conduit", "OST_DuctFitting", "OST_DuctCurves", "OST_DuctAccessory", "OST_FabricationParts", "OST_FabricationPipework", "OST_FabricationHangers", "OST_PipeAccessory", "OST_PipeFitting", "OST_PipeCurves", "OST_FabricationDuctwork" });
```
Comment (pabloderen) [inline RenumberParts_Shared/Views/MainForm.xaml.cs]: Change to categories

**Review by pabloderen — CHANGES_REQUESTED** [review]

File: RenumberParts_Shared/Views/ColorForm.xaml.cs [hunk 8]
```diff
@@ -18,13 +19,30 @@ namespace RenumberParts
     /// </summary>
     public partial class ColorForm : Window
     {
+        public System.Drawing.Color colorSelected;
+        public ColorForm()
```
Comment (ENGworks-DEV) [inline RenumberParts_Shared/Views/ColorForm.xaml.cs:23]: This does something?

**Review by ENGworks-DEV — DISMISSED** [review]

**Review by pabloderen — APPROVED** [review]

**Pull request merged**